File: src/gcov/State/EditorState.java
package gcov.State;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.components.State;
import com.intellij.openapi.project.Project;
import com.intellij.util.xmlb.XmlSerializerUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@State(name = "GCoverageShowInEditor")
public class EditorState implements PersistentStateComponent<EditorState> {

    public boolean showInEditor = true;

    public static EditorState getInstance(@NotNull Project project) {
        return ServiceManager.getService(project,EditorState.class);
    }

    @Nullable
    @Override
    public EditorState getState() {
        return this;
    }

    @Override
    public void loadState(@NotNull EditorState editorState) {
        XmlSerializerUtil.copyBean(editorState,this);
    }
}

File: src/gcov/state/EditorState.kt
package gcov.state

import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.ServiceManager
import com.intellij.openapi.components.State
import com.intellij.openapi.project.Project
import com.intellij.util.xmlb.XmlSerializerUtil

@State(name = "GCoverageShowInEditor")
class EditorState : PersistentStateComponent<EditorState> {

    var showInEditor = true

    override fun getState(): EditorState = this

    override fun loadState(editorState: EditorState) = XmlSerializerUtil.copyBean(editorState, this)

    companion object {

        fun getInstance(project: Project): EditorState = ServiceManager.getService(project, EditorState::class.java)!!
    }
}
